internal/provider/mcr_resource.go

```diff
@@ -0,0 +1,589 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	megaport "github.com/megaport/megaportgo"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &mcrResource{}
+	_ resource.ResourceWithConfigure   = &mcrResource{}
+	_ resource.ResourceWithImportState = &mcrResource{}
+)
+
+// mcrResourceModel maps the resource schema data.
+type mcrResourceModel struct {
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	ID                    types.Int64  `tfsdk:"product_id"`
+	UID                   types.String `tfsdk:"product_uid"`
```

Comment:
the port resource uses `uid` instead of `product_uid` we should probably keep them consistent

Reply:
Change pushed to have both the Port and MCR resources use `product_uid` to be consistent with the Megaport API.